Source/WebKit/UIProcess/WebsiteData/Cocoa/WebsiteDataStoreCocoa.mm

```diff
@@ -684,19 +684,20 @@ static NavigatingToAppBoundDomain schemeOrDomainIsAppBound(const String& host, c
 
 String WebsiteDataStore::resolvedContainerTemporaryDirectory()
 {
-    if (m_resolvedContainerTemporaryDirectory.isNull()) {
-        if (!isPersistent())
-            m_resolvedContainerTemporaryDirectory = emptyString();
-        else
-            m_resolvedContainerTemporaryDirectory = defaultContainerTemporaryDirectory();
-    }
+    if (m_resolvedContainerTemporaryDirectory.isNull())
+        m_resolvedContainerTemporaryDirectory = defaultResolvedContainerTemporaryDirectory();
 
     return m_resolvedContainerTemporaryDirectory;
 }
 
-String WebsiteDataStore::defaultContainerTemporaryDirectory()
+String WebsiteDataStore::defaultResolvedContainerTemporaryDirectory()
 {
-    return NSTemporaryDirectory();
+    static NeverDestroyed<String> resolvedTemporaryDirectory;
+    static std::once_flag once;
+    std::call_once(once, [] {
+        resolvedTemporaryDirectory.get() = resolveAndCreateReadWriteDirectoryForSandboxExtension(String(NSTemporaryDirectory()));
```

Comment:
Do we really need the explicit `String()`?

Reply:
Yes, we currently need that.

Thanks for reviewing!

Reply:
Nice, this should reduce some calls to resolve for different WebsiteDataStores

Reply:
Yes, I think this should reduce the number of resolve calls.

Thanks for reviewing!